realm/realm-library/src/main/java/io/realm/StandardRealmObjectSchema.java

```diff
@@ -676,6 +687,12 @@ private void checkFieldExists(String fieldName) {
         }
     }
 
+    private void checkAddPrimaryKeyForSync() {
```

Comment:
I totally think we should disallow adding primary for non-sync in 4.0. But we need to align with other bindings.